bgpd/bgpd.c

```diff
@@ -3994,6 +3994,18 @@ void bgp_free(struct bgp *bgp)
 			ecommunity_free(&bgp->vpn_policy[afi].rtlist[dir]);
 		if (bgp->vpn_policy[afi].tovpn_rd_pretty)
 			XFREE(MTYPE_BGP, bgp->vpn_policy[afi].tovpn_rd_pretty);
+		if (bgp->vpn_policy[afi].tovpn_sid_locator != NULL)
+			srv6_locator_chunk_free(
+				&bgp->vpn_policy[afi].tovpn_sid_locator);
+		if (bgp->vpn_policy[afi].tovpn_zebra_vrf_sid_last_sent != NULL)
+			XFREE(MTYPE_BGP_SRV6_SID,
+			      bgp->vpn_policy[afi]
+				      .tovpn_zebra_vrf_sid_last_sent);
+		if (bgp->vpn_policy[afi].tovpn_sid != NULL) {
+			sid_unregister(bgp, bgp->vpn_policy[afi].tovpn_sid);
```

Comment:
I agree with Donatas that SIDs should be unregistered before calling XFREE, but this is NOT the correct place to unregister them.

On line 3978, we delete the `bgp->srv6_functions list` (in `bgp_srv6_cleanup`). On line 4005, you call `sid_unregister`, which iterates over the `bgp->srv6_functions` list we just deleted.

Reply:
what if i move the ```bgp_srv6_cleanup``` to the line 4010 so that the ```sid_unregister``` can delete the ```bgp->srv6_functions``` before ```bgp->srv6_functions list``` is deleted. would that be suitable solution?

thank you for reviewing

Reply:
yes, that would be ok.

Reply:
yes, that would be ok